t/chaos/kill-etcd_test.go

```diff
@@ -0,0 +1,251 @@
+			}
+		}`,
+		ExpectStatus: expectStatus,
+	})
+}
+
+func getRoute(e *httpexpect.Expect, expectStatus int) {
+	caseCheck(httpTestCase{
+		E:            e,
+		Method:       http.MethodGet,
+		Path:         "/hello",
+		Headers:      map[string]string{"Host": "foo.com"},
+		ExpectStatus: expectStatus,
+	})
+}
+
+func deleteRoute(e *httpexpect.Expect, expectStatus int) {
+	caseCheck(httpTestCase{
+		E:            e,
+		Method:       http.MethodDelete,
+		Path:         "/apisix/admin/routes/1",
+		Headers:      map[string]string{"X-API-KEY": token},
+		ExpectStatus: expectStatus,
+	})
+}
+
+func getWithoutTest(g *WithT, path string, headers map[string]string) string {
+	client := &http.Client{}
+	url := host + path
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	g.Expect(err).To(BeNil())
+	for key, value := range headers {
+		req.Header.Add(key, value)
+	}
+
+	resp, err := client.Do(req)
+	g.Expect(err).To(BeNil())
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	g.Expect(err).To(BeNil())
+
+	return string(body)
+}
+
+func testPrometheusEtcdMetric(e *httpexpect.Expect, expectEtcd int) {
+	caseCheck(httpTestCase{
+		E:          e,
+		Method:     http.MethodGet,
+		Path:       "/apisix/prometheus/metrics",
+		ExpectBody: fmt.Sprintf("apisix_etcd_reachable %d", expectEtcd),
+	})
+}
+
+// get the first line which contains the key
+func getPrometheusMetric(g *WithT, key string) string {
+	body := getWithoutTest(g, "/apisix/prometheus/metrics", nil)
+	resps := strings.Split(body, "\n")
+	var targetLine string
+	for _, line := range resps {
+		if strings.Contains(line, key) {
+			targetLine = line
+			break
+		}
+	}
+	targetSlice := strings.Fields(targetLine)
+	g.Expect(len(targetSlice) == 2).To(BeTrue())
+	return targetSlice[1]
+}
+
+func getIngressBandwidthPerSecond(g *WithT) float64 {
+	key := "apisix_bandwidth{type=\"ingress\","
+	bandWidthString := getPrometheusMetric(g, key)
+	bandWidthStart, err := strconv.ParseFloat(bandWidthString, 64)
+	g.Expect(err).To(BeNil())
+
+	time.Sleep(5 * time.Second)
+	bandWidthString = getPrometheusMetric(g, key)
+	bandWidthEnd, err := strconv.ParseFloat(bandWidthString, 64)
+	g.Expect(err).To(BeNil())
+
+	return (bandWidthEnd - bandWidthStart) / 5.0
+}
+
+func runCommand(t *testing.T, cmd string) string {
+	out, err := exec.Command("bash", "-c", cmd).CombinedOutput()
+	if err != nil {
+		t.Fatalf("fail to run command %s: %s, %s", cmd, err.Error(), out)
+	}
+	return string(out)
+}
+
+func roughCompare(a float64, b float64) bool {
+	ratio := a / b
+	if ratio < 1.2 && ratio > 0.8 {
+		return true
+	}
+	return false
+}
+
+func TestGetSuccessWhenEtcdKilled(t *testing.T) {
+	g := NewWithT(t)
+	e := httpexpect.New(t, host)
+
+	// check if everything works
+	setRoute(e, http.StatusCreated)
+	getRoute(e, http.StatusOK)
+	testPrometheusEtcdMetric(e, 1)
+
+	// run in background
+	go func() {
+		for i := 1; ; i++ {
+			go getWithoutTest(g, "/hello", map[string]string{"Host": "foo.com"})
```

Comment:
**HELP NEEDED!!**

Here I try to visit the route in the same frequency, so I could check if it keep the same before and after etcd shutdown.

For some reason, using bare `net/http` pkg to access the route would always return "404 Route not found", although the route is created, and accessing the same URL with `httpexpect` could work(in line 207).

While the reason I don't want to use `httpexpect` this place, is due to it would print lots of log each time a request sent and it would make test log into a mess. So a bare HTTP get is needed.

Reply:
Hi @juzhiyuan could you offer some help on this

Reply:
Since using `httpexpect` would print log each time a request sent, I turn the route visit frequency from 10 times/s to 1 times/s to reduce the log printing. The reason not to reduce visit duration is that, when frequency is larger than 1 time/s, some visit would return `"503 Service Unavailable"`, although I don't think that's reasonable since the frequency is still very low compared to the real usage.

Reply:
The question is raised in https://github.com/apache/apisix-dashboard/issues/1388